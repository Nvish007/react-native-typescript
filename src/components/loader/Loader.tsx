import React from 'react';
import {View, ActivityIndicator, StyleSheet} from 'react-native';

const Loader: React.FC = () => {
  return (
    <View style={styles.container}>
      <ActivityIndicator
        testID="activity-indicator"
        size={'large'}
        color={'blue'}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default Loader;
