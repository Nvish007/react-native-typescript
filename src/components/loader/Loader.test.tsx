import React from 'react';
import {render} from '@testing-library/react-native';
import Loader from './Loader';

describe('Loader Component', () => {
  it('renders without crashing', () => {
    const {getByTestId} = render(<Loader />);
    const loaderComponent = getByTestId('activity-indicator');
    expect(loaderComponent).toBeTruthy();
  });

  it('renders ActivityIndicator with correct size and color', () => {
    const {getByTestId} = render(<Loader />);
    const activityIndicator = getByTestId('activity-indicator');
    expect(activityIndicator.props.size).toBe('large');
    expect(activityIndicator.props.color).toBe('blue');
  });

  it('matches snapshot', () => {
    const {toJSON} = render(<Loader />);
    expect(toJSON()).toMatchSnapshot();
  });
});
