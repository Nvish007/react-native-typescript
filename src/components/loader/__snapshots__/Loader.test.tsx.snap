// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    color="blue"
    size="large"
    testID="activity-indicator"
  />
</View>
`;
